src/nomad_simulations/schema_packages/utils/libxc/registry.py

```diff
@@ -0,0 +1,41 @@
+import json
+from functools import lru_cache
+from pathlib import Path
+
+# Path to the registry JSON file
+_REGISTRY_PATH = Path(__file__).parent / 'xc_registry_min.json'
+
+
+def _keyize(s: str) -> str:
+    # "XC_GGA_X_PBE" -> "GGAXPBE"
+    return (
+        s.strip()
+        .upper()
+        .replace('XC_', '')
+        .replace('_', '')
+        .replace('-', '')
+        .replace(' ', '')
+    )
+
+
+@lru_cache(maxsize=1)
+def _load_index() -> tuple[dict[str, dict], dict[int, dict], dict[str, dict]]:
```

Comment:
you might want to use nomad.archive.storage_v2.ArchiveReader and ArchiveWriter, in a separate pr maybe if the database is large. it is meant for archive dict but i think it could be used for any json. so the data is lazily loaded.